pkg/cloudprovider/external_test.go

```diff
@@ -167,6 +167,13 @@ func TestIsCloudProviderExternal(t *testing.T) {
 		},
 		featureGate: nil,
 		expected:    true,
+	}, {
+		name: "No FeatureGate, Platform: Nutanix",
+		status: &configv1.PlatformStatus{
+			Type: configv1.NutanixPlatformType,
+		},
+		featureGate: nil,
+		expected:    true,
```

Comment:
Please also add a test that shows what happens when there is a feature gate

Reply:
Added the test cases.